extensions/refresh-catalog/src/main/java/org/eclipse/dataspaceconnector/mvd/RegistrationServiceNodeDirectoryExtension.java

```diff
@@ -15,26 +15,28 @@
 package org.eclipse.dataspaceconnector.mvd;
 
 import org.eclipse.dataspaceconnector.catalog.spi.FederatedCacheNodeDirectory;
+import org.eclipse.dataspaceconnector.common.configuration.ConfigurationFunctions;
+import org.eclipse.dataspaceconnector.registration.client.ApiClientFactory;
+import org.eclipse.dataspaceconnector.registration.client.api.RegistryApi;
 import org.eclipse.dataspaceconnector.spi.system.Provides;
 import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
 import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
 import org.eclipse.dataspaceconnector.spi.types.TypeManager;
 
-import java.nio.file.Path;
-import java.util.Objects;
-
 /**
- * Extension to set up federated cache directory using a set of JSON files as backend.
+ * Extension to set up federated cache directory using Registration Service API as backend.
  */
 @Provides(FederatedCacheNodeDirectory.class)
-public class JsonFileBasedNodeDirectoryExtension implements ServiceExtension {
+public class RegistrationServiceNodeDirectoryExtension implements ServiceExtension {
+
+    static final String API_URL = "http://localhost:8181/api";
+
     @Override
     public void initialize(ServiceExtensionContext context) {
         var monitor = context.getMonitor();
         TypeManager typeManager = context.getTypeManager();
-        var nodeJsonPath = Path.of(Objects.requireNonNull(System.getenv("NODES_JSON_DIR"), "Env var NODES_JSON_DIR is null"));
-        var nodeJsonPrefix = Objects.requireNonNull(System.getenv("NODES_JSON_FILES_PREFIX"), "Env var NODES_JSON_FILES_PREFIX is null");
-        var service = new JsonFileBasedNodeDirectory(nodeJsonPath, nodeJsonPrefix, monitor, typeManager);
+        var registrationServiceApiUrl = ConfigurationFunctions.propOrEnv("REGISTRATION_SERVICE_API_URL", API_URL);
```

Comment:
```suggestion
        var registrationServiceApiUrl = ConfigurationFunctions.propOrEnv("registration.service.api.url", API_URL);
```

should automatically map to `REGISTRATION_SERVICE_API_URL` env var